### ISSUE TITLE
Interface is awkward to work with

### ISSUE THREAD

**Issue by chrismytton** [everypolitician/combine_popolo_memberships#7]
Trying to figure out how this works, even after reading the docs/tests/implementation, is quite challenging! I think a big part of this is that the interface isn't very intuitive, it looks like it uses keyword arguments, but actually it's just a Hash where the keys represent where the `id` should be copied to in the output.

Additionally, the code is just two functions packaged into a module at the moment, it could probably benefit from using classes and instances.

**Comment by tmtmtmtm** [comment]
Yep, this was originally just a quick extraction from a scraper, and it's served us well enough, but we could almost certainly do with something that makes more sense (and isn't even conceptually coupled to Popolo).

NB: Most of the date handling stuff shouldn't really live in this at all — I've always missed a good Ruby equivalent of [Date::Range](http://search.cpan.org/~tmtm/Date-Range-1.41/lib/Date/Range.pm)

**Comment by chrismytton** [comment]
> NB: Most of the date handling stuff shouldn't really live in this at all — I've always missed a good Ruby equivalent of Date::Range

👍 I've opened https://github.com/everypolitician/combine_popolo_memberships/issues/8 to track moving the date handling bits out.
